resources/providers/gcplib/inventory/provider.go

```diff
@@ -199,6 +210,50 @@ func (p *Provider) Close() error {
 	return p.inventory.Close()
 }
 
+// EnrichNetworkAssets enriches the network assets with dns policy if exists
+func (p *Provider) EnrichNetworkAssets(assets []*ExtendedGcpAsset) []*ExtendedGcpAsset {
+	if len(assets) == 0 {
+		p.log.Infof("no %s assets were listed", ComputeNetworkAssetType)
+		return assets
+	}
+
+	dnsPolicyAssets := getAllAssets(p.log, p.inventory.ListAssets(p.ctx, &assetpb.ListAssetsRequest{
+		Parent:      p.config.Parent,
+		AssetTypes:  []string{DnsPolicyAssetType},
+		ContentType: assetpb.ContentType_RESOURCE,
+	}))
+
+	if len(dnsPolicyAssets) == 0 {
+		p.log.Infof("no %s assets were listed, return original assets", DnsPolicyAssetType)
+		return assets
+	}
+
+	p.log.Infof("attempting to enrich %d %s assets with dns policy", len(assets), ComputeNetworkAssetType)
+
+outerLoop:
+	for i := 0; i < len(assets); i++ {
```

Comment:
please use `for _, networkAssetFields := range assets`

Reply:
@olegsu - I want to use the loop scope - `outerLoop `, this is why I've used the classic for loop.